fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlResourceDAO.java

```diff
@@ -261,14 +252,11 @@ public Integer readResourceTypeId(String resourceType) throws FHIRPersistenceDBC
                 logger.finer("DB read resource type id complete. executionTime=" + dbCallDuration + "ms");
             }
             resourceTypeId = stmt.getInt(2);
-        } catch(FHIRPersistenceDBConnectException e) {
-            throw e;
         } catch (Throwable e) {
             final String errMsg = "Failure storing Resource type name id: name=" + resourceType;
             FHIRPersistenceDataAccessException fx = new FHIRPersistenceDataAccessException(errMsg);
             throw severe(logger, fx, e);
         } finally {
-            this.cleanup(stmt, connection);
```

Comment:
Where does this happen now?

Reply:
Connections are opened and closed using try-with-resource in the corresponding FHIRPersistenceJDBCImpl method. We keep connection management in one place. Much less risk.

Reply:
I've added a cleanup(stmt). The statement would be closed as soon as the connection (wrapper) is closed, but it makes sense to close it where it was opened. We ought to use try-with-resource for all statements, but doing so requires a bit more finesse in the exception handling.